tests/test_entity.py

```diff
@@ -23,14 +33,17 @@ def test_value_setting(self):
     def test_invalid_values(self):
         entity = TestEntity()
         # Test setting invalid values
-        self.assertRaises(InvalidValueError, setattr, entity, "prop1", 10)  # Text property setting int
-        self.assertRaises(InvalidValueError, setattr, entity, "prop2", "text")  # Int property setting Text
-        self.assertRaises(InvalidValueError, setattr, entity, "prop2", 10.3)  # Int property setting Float
+        message = "10 is not a valid value for property prop1 of type TextProperty"
+        self.assertRaisesWithMessage(InvalidValueError, message, setattr, entity, "prop1", 10)  # Text property setting int
```

Comment:
Minor: isn't this clearer?
```suggestion
        self.assertRaisesWithMessage(InvalidValueError, message, lambda: entity.prop1 = 10)  # Text property setting int
```